### PR TITLE
Add focused styling to buttons

### PR DESCRIPTION
**Pull request by rubenwardy**
Part of: #11392

Currently, it's not possible to see whether a button is focused. This adds a styling state that allows focused buttons to be styled, in the future focus will be rendered by default.

This is a prerequiste for making it easier to use Minetest with a keyboard only. This will also be used by gamepads in the future as well.

Currently focus styling will always be visible, but it's possible that game developers will only want to show focus styling when the user tabs. Browsers distinguish between this using `:focus` and `:focus-visible` selectors, the later only applying when the user has tabbed or is using their keyboard for navigation. I haven't implemented the latter, that's something for another time

A good next PR will be adding an outline property which can be used for showing focus

```
style[one_btn14:focused;textcolor=red]
image_button[0,9.6;1,1;testformspec_button_image.png;one_btn14;Bg]
```

## To do

This PR is Ready for Review.

## How to test

`/test_formspec` - "Bg" and "9-slice bg" buttons at the bottom

image

### CODE REVIEW

**Comment by rubenwardy** [comment]
> A good next PR will be adding an outline property which can be used for showing focus

Wondering how best to do this without having to modify every element. Perhaps a new GUIOutlineBox element that wraps the added element and then renders based on the style, would only need to wrap all elements that have a modified `outline` style property then

**Comment by SmallJoker** [comment]
> Wondering how best to do this without having to modify every element.

I am not sure whether I understand your problem correctly, but wouldn't a general `*` match in `style_type[]` be convenient for that? Elements that cannot be focused (or where it makes no sense) would not have the focus property in the first place.

**Comment by rubenwardy** [comment]
That's talking about how to render outlines on all elements in C++. In the style API, yes you'd just use `*`. That's an unrelated thought anyway, probably shouldn't have distracted this PR with it

File: src/gui/StyleSpec.h [hunk 1]
```diff
@@ -154,6 +157,8 @@ class StyleSpec
 			return STATE_HOVERED;
 		} else if (name == "pressed") {
 			return STATE_PRESSED;
+		} else if (name == "focused") {
+			return STATE_FOCUSED;
```
Comment (v-rob) [inline src/gui/StyleSpec.h:161]: It seems preferable to me to keep these if statements in the same order as the `State` enum.

**Review by v-rob — APPROVED** [review]
Tested and code looks fine

**Pull request merged**